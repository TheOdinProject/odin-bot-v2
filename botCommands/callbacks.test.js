const command = require('./callbacks');

describe('?callbacks', () => {
  it.each([
    ['?callbacks'],
    [' ?callbacks'],
    ['?callbacks @odin-bot'],
    [' ?callbacks @odin-bot'],
  ])("'%s' triggers the callback", (string) => {
    expect(command.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['?coolbacks'],
    ['callbacks'],
    ['?cllbacks'],
    ['?callback'],
    ['?call backs'],
    ['```function("?call backs", () => {}```'],
    ['?CallBacks'],
    [''],
    [' '],
    ['? '],
    ['@odin-bot ? callbacks'],
    ['?call&ba^cks'],
    ['?callba^cks'],
    ['@odin-bot? callbacks'],
  ])("'%s' does not trigger the callback", (string) => {
    expect(command.regex.test(string)).toBeFalsy();
  });

  it.each([
    ['how about dem ?callbacks'],
    ['i think ?callbacks are great'],
    ['hey @odin-bot, teach me ?callbacks'],
    ['?@odin-bot ^ ?me ?callbacks ?test$*'],
  ])("'%s' - the command can be anywhere in the string", (string) => {
    expect(command.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['@user?callbacks'],
    ['watchaknowabout?callbacks'],
    ['?callbackssgrealkdmsfalnd'],
    ['?callbacks?'],
    ['?callbacks*'],
    ['?callbacks...'],
  ])(
    "'%s' - command should be its own word/group - no leading or trailing characters",
    (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    },
  );
});

describe('?callbacks callback', () => {
  it('should return the correct output', () => {
    expect(command.cb()).toMatchSnapshot();
  });
});
