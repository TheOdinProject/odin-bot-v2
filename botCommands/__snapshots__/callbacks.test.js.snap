// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`?callbacks callback should return the correct output 1`] = `"**DID SOMEONE SAY CALLBACKS?: https://briggs.dev/blog/understanding-callbacks**"`;
